src/main/kotlin/com/cognifide/gradle/aem/pkg/DownloadTask.kt

```diff
@@ -0,0 +1,94 @@
+package com.cognifide.gradle.aem.pkg
+
+import com.cognifide.gradle.aem.api.AemConfig
+import com.cognifide.gradle.aem.api.AemNotifier
+import com.cognifide.gradle.aem.api.AemTask
+import com.cognifide.gradle.aem.base.vlt.CheckoutConfig
+import com.cognifide.gradle.aem.base.vlt.VltException
+import com.cognifide.gradle.aem.instance.InstanceSync
+import com.cognifide.gradle.aem.internal.PropertyParser
+import com.cognifide.gradle.aem.internal.file.FileContentReader
+import org.gradle.api.file.DuplicatesStrategy
+import org.gradle.api.tasks.*
+import org.gradle.api.tasks.bundling.Zip
+
+open class DownloadTask : Zip(), AemTask {
+
+    companion object {
+        const val NAME = "aemDownload"
+        const val CLASSIFIER_DOWNLOAD = "download"
+    }
+
+    @Nested
+    final override val config = AemConfig.of(project)
+
+    @InputDirectory
+    val prepareDir = AemTask.taskDir(project, PrepareTask.NAME)
+
+    @OutputDirectory
+    val downloadDir = AemTask.temporaryDir(project, NAME, CLASSIFIER_DOWNLOAD)
+
+    @Internal
+    val props = PropertyParser(project)
+
+    private val checkoutConfig = CheckoutConfig(project, props, config)
```

Comment:
I am opting about removing that another type of config. aem config and task internals should be enough ways for providing configuration.

Reply:
It used specifically to reuse configuration behavior between aemCheckout and aemDownload

Reply:
I dislike such relation